Builtin class "range" is being masked by reassignment

Python has a default class `range` ([ref](https://docs.python.org/3/library/stdtypes.html#range)), but in various places we are reassigning to that name and overwriting it with another value. But in other places, we use `range()` also. Possibly they haven't come into conflict so far, but it may be best to use another name so we don't run into a problem later.

```
❯ python3
Python 3.11.6 (main, Oct  2 2023, 13:45:54) [Clang 15.0.0 (clang-1500.0.40.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> type(range)
<class 'type'>
>>> range = 3
>>> type(range)
<class 'int'>
```

```
src/guiguts/maintext.py:394:    def do_select(self, range: IndexRange) -> None:
src/guiguts/maintext.py:400:        self.tag_add("sel", range.start.index(), range.end.index())
src/guiguts/maintext.py:438:        for range in ranges:
src/guiguts/maintext.py:439:            start = range.start.index()
src/guiguts/maintext.py:440:            end = range.end.index()
src/guiguts/maintext.py:444:            if range.end.row > range.start.row and range.end.col == 0:
src/guiguts/maintext.py:666:        range: IndexRange,
src/guiguts/maintext.py:684:        start_index = range.start.index()
src/guiguts/maintext.py:685:        stop_index = range.end.index()
src/guiguts/search.py:240:        range = None
src/guiguts/search.py:245:                range = sel_ranges[0]
src/guiguts/search.py:248:            range = IndexRange(maintext().start(), maintext().end())
src/guiguts/search.py:249:        if range:
src/guiguts/search.py:252:                range,
src/guiguts/search.py:271:        range = None
src/guiguts/search.py:274:                range = sel_ranges[0]
src/guiguts/search.py:276:            range = IndexRange(maintext().start(), maintext().end())
src/guiguts/search.py:277:        if range:
src/guiguts/search.py:280:                range,
```

pylint can:

```
src/guiguts/maintext.py:422:24: W0622: Redefining built-in 'range' (redefined-builtin)
```

Although, it ouputs a lot of other stuff, including "line too long" and stuff we would ignore since we're using black. I'll see if I can integrate pylint and quiet down things that don't matter... I'll open another issue on that.

pylint's current report for redefined built-ins:

```
❯ poetry run pylint ./src | grep built-in
src/guiguts/maintext.py:422:24: W0622: Redefining built-in 'range' (redefined-builtin)
src/guiguts/maintext.py:466:12: W0622: Redefining built-in 'range' (redefined-builtin)
src/guiguts/maintext.py:702:8: W0622: Redefining built-in 'range' (redefined-builtin)
src/guiguts/checkers.py:23:34: W0622: Redefining built-in 'range' (redefined-builtin)
src/guiguts/checkers.py:64:8: W0622: Redefining built-in 'range' (redefined-builtin)
src/guiguts/application.py:187:8: W0622: Redefining built-in 'help' (redefined-builtin)
src/guiguts/search.py:240:8: W0622: Redefining built-in 'range' (redefined-builtin)
src/guiguts/search.py:271:8: W0622: Redefining built-in 'range' (redefined-builtin)
```

--- a/src/guiguts/application.py
+++ b/src/guiguts/application.py
@@ -190,7 +190,7 @@ class Guiguts:
 
     def help_about(self, *args: Any) -> None:
         """Display a 'Help About' dialog."""
-        help = """Guiguts - an application to support creation of ebooks for PG
+        help_message = """Guiguts - an application to support creation of ebooks for PG
 
 Copyright Contributors to the Guiguts-py project.
 
@@ -211,7 +211,7 @@ License along with this program; if not, write to the
 Free Software Foundation, Inc., 51 Franklin Street,
 Fifth Floor, Boston, MA 02110-1301 USA."""
 
-        messagebox.showinfo(title="About Guiguts", message=help)
+        messagebox.showinfo(title="About Guiguts", message=help_message)
 
     def show_page_details_dialog(self, *args: Any) -> None:
         """Show the page details display/edit dialog."""

--- a/src/guiguts/maintext.py
+++ b/src/guiguts/maintext.py
@@ -434,13 +434,13 @@ class MainText(tk.Text):
         """Clear any current text selection."""
         self.tag_remove("sel", "1.0", tk.END)
 
-    def do_select(self, range: IndexRange) -> None:
+    def do_select(self, _range: IndexRange) -> None:
         """Select the given range of text.
 
         Args:
             IndexRange containing start and end of text to be selected."""
         self.clear_selection()
-        self.tag_add("sel", range.start.index(), range.end.index())
+        self.tag_add("sel", _range.start.index(), _range.end.index())
 
     def selected_ranges(self) -> list[IndexRange]:
         """Get the ranges of text marked with the `sel` tag.
@@ -488,8 +488,8 @@ class MainText(tk.Text):
         """Delete the selected column text."""
         if not (ranges := self.selected_ranges()):
             return
-        for range in ranges:
-            self.delete(range.start.index(), range.end.index())
+        for _range in ranges:
+            self.delete(_range.start.index(), _range.end.index())
 
     def column_copy_cut(self, cut: bool = False) -> None:
         """Copy or cut the selected text to the clipboard.
@@ -502,9 +502,9 @@ class MainText(tk.Text):
         if not (ranges := self.selected_ranges()):
             return
         self.clipboard_clear()
-        for range in ranges:
-            start = range.start.index()
-            end = range.end.index()
+        for _range in ranges:
+            start = _range.start.index()
+            end = _range.end.index()
             string = self.get(start, end)
             if cut:
                 self.delete(start, end)
@@ -787,7 +787,7 @@ class MainText(tk.Text):
     def find_matches(
         self,
         search_string: str,
-        range: IndexRange,
+        text_range: IndexRange,
         nocase: bool,
         regexp: bool,
         wholeword: bool,
@@ -796,7 +796,7 @@ class MainText(tk.Text):
 
         Args:
             search_string: String/regex to be searched for.
-            range: Range in which to search.
+            text_range: Range in which to search.
             nocase: True to ignore case.
             regexp: True if string is a regex; False for exact string match.
             wholeword: True to only search for whole words (i.e. word boundary at start & end).
@@ -805,8 +805,8 @@ class MainText(tk.Text):
             List of FindMatch objects, each containing index of start and count of characters in a match.
             Empty list if no matches.
         """
-        start_index = range.start.index()
-        stop_index = range.end.index()
+        start_index = text_range.start.index()
+        stop_index = text_range.end.index()
         if regexp:
             search_string = convert_to_tcl_regex(search_string)
         if wholeword:

